sklearn/semi_supervised/tests/test_self_training.py

```diff
@@ -0,0 +1,262 @@
+import numpy as np
+import pytest
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_raise_message
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_warns_message
+from sklearn.exceptions import NotFittedError, ConvergenceWarning
+
+from sklearn.semi_supervised import SelfTrainingClassifier
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.svm import SVC
+from sklearn.model_selection import train_test_split
+from sklearn.datasets import load_iris
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.metrics import accuracy_score
+
+# Author: Oliver Rausch <rauscho@ethz.ch>
+# License: BSD 3 clause
+
+# load the iris dataset and randomly permute it
+iris = load_iris()
+X_train, X_test, y_train, y_test = train_test_split(iris.data,
+                                                    iris.target,
+                                                    random_state=0)
+
+n_labeled_samples = 50
+
+y_train_missing_labels = y_train.copy()
+y_train_missing_labels[n_labeled_samples:] = -1
+mapping = {0: 'A', 1: 'B', 2: 'C', -1: '-1'}
+y_train_missing_strings = np.vectorize(mapping.get)(
+    y_train_missing_labels).astype(object)
+y_train_missing_strings[y_train_missing_labels == -1] = -1
+
+lb = LabelBinarizer()
+y_train_missing_dummies = lb.fit_transform(y_train_missing_labels)
+
+
+def test_missing_predict_proba():
+    # Check that an error is thrown if predict_proba is not implemented
+    base_classifier = SVC(probability=False, gamma='scale')
+    self_training = SelfTrainingClassifier(base_classifier)
+    message = "base_classifier (SVC) should implement predict_proba!"
+    assert_raise_message(ValueError, message, self_training.fit, X_train,
```

Comment:
We are moving to `pytest.raises` for exceptions.

Reply:
Do you mean like this? 
```python
with pytest.raises(ValueError) as e:
    message = "base_classifier (SVC) should implement predict_proba!"
    self_training.fit(X_train, y_train_missing_labels)

assert e.value.message == message
```
If yes, i'll update the others

Reply:
```
assert e.value.message == message
```
should be
```
assert e.value.args[0] == message
```

Reply:
The code base uses `pytest.raises(..., match=)` and use a regex. To make this easier, we could change the error message to "base_classifier, SVC, should implement predict_proba"